### PR TITLE
Fixed #34609 -- Deprecated calling format_html() without arguments.

### PR DESCRIPTION
**Pull request by DevilsAutumn**
[Ticket #34609](https://code.djangoproject.com/ticket/34609)

### CODE REVIEW

File: django/utils/html.py [hunk 1]
```diff
@@ -100,6 +102,11 @@ def format_html(format_string, *args, **kwargs):
     and call mark_safe() on the result. This function should be used instead
     of str.format or % interpolation to build up small HTML fragments.
     """
+    if not (args or kwargs):
+        warnings.warn(
+            "Calling format_html() without args or kwargs is depricated.",
```
Comment (felixxm) [inline django/utils/html.py]:
```suggestion
            "Calling format_html() without args or kwargs is deprecated.",
```

File: docs/internals/deprecation.txt [hunk 2]
```diff
@@ -35,6 +35,8 @@ details on these changes.
 * The default scheme for ``forms.URLField`` will change from ``"http"`` to
   ``"https"``.
 
+* Support for calling format_html() without args or kwargs will be removed.
```
Comment (felixxm) [inline docs/internals/deprecation.txt:38]: This should be documented in `format_html()` docs and release notes.

File: django/utils/html.py [hunk 3]
```diff
@@ -100,6 +102,11 @@ def format_html(format_string, *args, **kwargs):
     and call mark_safe() on the result. This function should be used instead
     of str.format or % interpolation to build up small HTML fragments.
     """
+    if not (args or kwargs):
```
Comment (felixxm) [inline django/utils/html.py:105]: Add a comment how it should be changed when the deprecation ends, e.g.:
```
# RemovedInDjango60Warning: when the deprecation ends, replace with:
#     raise ValueError(...)
```

**Review by felixxm — COMMENTED** [review]
@DevilsAutumn Thanks :+1:

File: tests/utils_tests/test_html.py [hunk 4]
```diff
@@ -65,6 +66,12 @@ def test_format_html(self):
             "&lt; Dangerous &gt; <b>safe</b> &lt; dangerous again <i>safe again</i>",
         )
 
+    def test_format_html_no_params(self):
+        msg = "Calling format_html() without args or kwargs is deprecated."
+        with self.assertWarnsMessage(RemovedInDjango60Warning, msg):
+            name = "Adam"
```
Comment (adamchainz) [inline tests/utils_tests/test_html.py:72]: hello

File: docs/internals/deprecation.txt (hunk 2 above)
Comment (adamchainz) [inline docs/internals/deprecation.txt:38]: Also:

```suggestion
* Support for calling ``format_html()`` without any arguments will be removed.
```

**Review by adamchainz — CHANGES_REQUESTED** [review]
Looking great, thanks for picking this up. A few small suggestions from me.

Can we retitle the commit and PR:

```
Fixed #34609 -- Deprecated calling format_html() without arguments.
```

I think we can ignore the distinction between args and kwargs when describing the change.

File: tests/utils_tests/test_html.py (hunk 4 above)
Comment (DevilsAutumn) [reply tests/utils_tests/test_html.py:72]: Thought you'll like it. 😝

**Review by felixxm — COMMENTED** [review]
@DevilsAutumn Thanks :+1:

I pushed tiny edits to docs.

File: django/utils/html.py [hunk 5]
```diff
@@ -100,6 +102,13 @@ def format_html(format_string, *args, **kwargs):
...
+        # RemovedInDjango60Warning: when the deprecation ends, replace with:
+        # raise ValueError("args or kwargs must be provided.")
```
Comment (smithdc1) [inline django/utils/html.py:107]: For my own understanding. Why this approach over changing the function signature?

Comment (felixxm) [reply django/utils/html.py:107]: How you would change it? (positional or keyword arguments must be provided besides `format_string`) :thinking:

**Pull request merged**